xbmc/interfaces/json-rpc/FileOperations.cpp

```diff
@@ -213,6 +214,41 @@ JSONRPC_STATUS CFileOperations::GetFileDetails(const std::string &method, ITrans
   return OK;
 }
 
+JSONRPC_STATUS CFileOperations::SetFileDetails(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  std::string file = parameterObject["file"].asString();
+  if (!CFile::Exists(file))
+    return InvalidParams;
+
+  if (!CFileUtils::RemoteAccessAllowed(file))
+    return InvalidParams;
+
+  CVideoDatabase videodatabase;
+  if (!videodatabase.Open())
+    return InternalError;
+
+  int fileId = videodatabase.AddFile(file);
+
+  CVideoInfoTag infos;
+  if (!videodatabase.GetFileInfo("", infos, fileId))
+    return InvalidParams;
+
+  if (parameterObject.isMember("playcount") || parameterObject.isMember("lastplayed"))
```

Comment:
This check is not necessary. The request handler in combination with the API definition will make sure that the parameters are there (but their value might be `null`).